src/polymer/behavior-scanner.ts

```diff
@@ -27,24 +27,16 @@ import {declarationPropertyHandlers, PropertyHandlers} from './declaration-prope
 import * as docs from './docs';
 import {getOrInferPrivacy, toScannedPolymerProperty} from './js-utils';
 
-interface KeyFunc<T> {
-  (value: T): any;
-}
-
-function dedupe<T>(array: T[], keyFunc: KeyFunc<T>): T[] {
-  const bucket = {};
+function dedupe<T, K>(array: T[], keyFunc: (value: T) => K): T[] {
```

Comment:
true, though I hope to delete this by just turning these arrays into maps anyways